### ISSUE TITLE
[Bug]: Som Queries error when run tpch test with 10G scale

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#6178]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):a17ab3ee2026c1a4173e8ea8f44000eae59abc17
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

TPCH QUERY RESULTS:

Now start to execute the query queries/q12.sql,please wait.....
TThe query queries/q12.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648
Now start to execute the query queries/q13.sql,please wait.....

Now start to execute the query queries/q18.sql,please wait.....
TThe query queries/q18.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648

Now start to execute the query queries/q21.sql,please wait.....
TThe query queries/q21.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648

ow start to execute the query queries/q4.sql,please wait.....
TThe query queries/q4.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648

Now start to execute the query queries/q7.sql,please wait.....
TThe query queries/q7.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648

Now start to execute the query queries/q9.sql,please wait.....
TThe query queries/q9.sql  has failed to  been executed.
ERROR 20101 (HY000) at line 1: internal error: Invalid alloc size 2147483648

MO  error log:
2022/11/01 22:56:08.655196 +0800 DEBUG gossip registry/gossip_logger.go:38 memberlist: Stream connection from=127.0.0.1:46568
2022/11/01 22:56:08.936602 +0800 INFO mpool/mpool.go:505 error: internal error: Invalid alloc size 2147483648 {"span": {}}
2022/11/01 22:56:08.936846 +0800 INFO vm/vm.go:57 error: internal error: panic runtime error: index out of range [66807912] with length 0:
runtime.goPanicIndexU
        /usr/local/go/src/runtime/panic.go:119
github.com/matrixorigin/matrixone/pkg/container/hashtable.(*StringHashMap).findCell
        /data1/sudong/mo-regression/matrixone/pkg/container/hashtable/string_hash_map.go:318
github.com/matrixorigin/matrixone/pkg/container/hashtable.(*StringHashMap).FindStringBatch
        /data1/sudong/mo-regression/matrixone/pkg/container/hashtable/string_hash_map.go:263
github.com/matrixorigin/matrixone/pkg/common/hashmap.(*strHashmapIterator).Find
        /data1/sudong/mo-regression/matrixone/pkg/common/hashmap/iterator.go:34
github.com/matrixorigin/matrixone/pkg/sql/colexec/join.(*container).probe
        /data1/sudong/mo-regression/matrixone/pkg/sql/colexec/join/join.go:135
github.com/matrixorigin/matrixone/pkg/sql/colexec/join.Call
        /data1/sudong/mo-regression/matrixone/pkg/sql/colexec/join/join.go:76
github.com/matrixorigin/matrixone/pkg/vm.Run
        /data1/sudong/mo-regression/matrixone/pkg/vm/vm.go:61
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).MergeRun
        /data1/sudong/mo-regression/matrixone/pkg/vm/pipeline/pipeline.go:147
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun
        /data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:110
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func2
        /data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:81
runtime.goexit

**Comment by m-schen** [comment]
need to modify the implement of hash table.
I will do it.

**Comment by m-schen** [comment]
fixed

**Comment by fengttt** [comment]
Did you verify with tpch 10g?

**Comment by aressu1985** [comment]
this error is fixed in commit:e23e00f1d01d43e33c509a6d239ccc38b5f93409

bug Q10 is still hung , and tracked by #6324

this issue can be closed
